iree/task/executor.c

```diff
@@ -116,6 +107,17 @@ iree_status_t iree_task_executor_create(
         &executor->transient_task_pool);
   }
 
+  // Wait handling polling and waiting use a dedicated thread to ensure that
+  // blocking syscalls stay off the workers.
```

Comment:
This could increase overheads in some environments... any concerns about that? In the web work I'm doing now, I'm wondering about Web Worker startup time then communication overhead between threads. Since we use shared memory, atomics, and other primitives (rather than messages) for communication, that overhead shouldn't change with an extra thread, right?

Reply:
In theory if the translated wasm has been cached then instantiating a worker shouldn't be too bad - it's not even allocating a big hunk of memory as it's all shared. Once running all communications are done via futex so no browser APIs involved thankfully.

For wait handles on web I was thinking of a wait_handle_inproc or something that models wait handles as tiered futices - then we can still use the same wait handle interfaces but not need any emscripten/posix support (no poll and such).

Reply:
I may do that (wait_handle_inproc) now actually - will help you when you get to it!